### PR TITLE
Submission table fixes

### PR DESCRIPTION
**Pull request by melneubert**
Fixed search and filtering for submissions table. Also added a filter for submissions in the remark state.

### CODE REVIEW

File: app/views/submissions/_submissions_table.js.jsx.erb [hunk 1]
```diff
@@ -102,6 +102,13 @@
             func: function(submission) {
               return submission.state == 'released';
             }
+          },
+          {
+            name: 'remark_requested',
+            text: '<%= j raw I18n.t(:'browse_submissions.show_remark_request') %>',
```
Comment (david-yz-liu) [inline app/views/submissions/_submissions_table.js.jsx.erb]: Use `t`, not `I18n.t`

File: app/views/submissions/_submissions_table.js.jsx.erb [hunk 2]
```diff
@@ -188,9 +195,9 @@
         var s = {};
         s['id'] = submission.id;
         if (submission.name_url === '') {
-          s['group_name'] = <span>{submission.name}</span>;
+          s['group_name'] = <span dangerouslySetInnerHTML={{__html: submission.name}} />;
```
Comment (david-yz-liu) [inline app/views/submissions/_submissions_table.js.jsx.erb]: Don't use this. Find another way to solve this problem.

Comment (melneubert) [inline app/views/submissions/_submissions_table.js.jsx.erb]: That's the attribute it was looking for in javascripts/ReactComponents/table.js. I guess I should be changing what it's looking for there as well?

Comment (david-yz-liu) [inline app/views/submissions/_submissions_table.js.jsx.erb]: Yes, that's exactly right.

**Comment by melneubert** [comment]
@david-yz-liu I made those changes you requested. I did some research and found that Reacts escapes all strings displayed, so I think that should fix the possibility that we were running into using dangerouslySetInnerHTML. 

I removed I18n.t in the place you requested, but I still see it in other places in this file, did you want me to change all instances or leave the other ones?

**Comment by david-yz-liu** [comment]
@melneubert Well done, looks great. I'll worry about the other I18n's later.

**Pull request merged**